entry.go

```diff
@@ -126,43 +128,33 @@ func (te *TunaEntry) Start() {
 			var np *nkn.NanoPay
 			for {
 				time.Sleep(DefaultNanoPayUpdateInterval)
-				bytesIn := atomic.LoadUint64(&te.bytesIn)
-				bytesOut := atomic.LoadUint64(&te.bytesOut)
-				entryToExitPrice, exitToEntryPrice := te.GetPrice()
-				delta := exitToEntryPrice*common.Fixed64(bytesIn-te.bytesInPaid)/TrafficUnit + entryToExitPrice*common.Fixed64(bytesOut-te.bytesOutPaid)/TrafficUnit
-				if delta == 0 {
-					continue
-				}
-				paymentReceiver := te.GetPaymentReceiver()
-				if np == nil || np.Address() != paymentReceiver {
-					var err error
-					np, err = te.Wallet.NewNanoPay(paymentReceiver, te.config.NanoPayFee, DefaultNanoPayDuration)
-					if err != nil {
-						continue
-					}
-				}
-				tx, err := np.IncrementAmount(delta.String())
+				session, err := te.getSession(false)
```

Comment:
I think it's better to get session after computing delta (like the original code) so we don't need to get or create session if there is no use.